### PR TITLE
Popup widget

### LINKED ISSUE #924: Create Popup widget to display widget popups consistently.
**Enhancement** 

We should create a `popup` widget similar to [`popper`](https://popper.js.org/).
This can be used by select / typeahead etc to position it's menu consistently.
Initially we should support `above` / `below` positioning and reposition the popup contents if required to ensure it will fit on the screen.

Additionally, we could support `left` / `right` as a follow up to enable tooltip / sub-menu style displays.

### PR DESCRIPTION
**Pull request by tomdye**
**Type:** feature

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/framework/blob/master/STYLE.md)
* [x] Unit tests are included in the PR
* [x] Any widget variant uses `theme.compose` like [this](https://github.com/dojo/widgets/issues/847)
* [x] WidgetProperties are exported

**Description:**

Adds popup widget and tests

Resolves #924

### CODE REVIEW

File: src/examples/src/widgets/popup/SetWidth.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+	return (
+		<Popup position="below" matchWidth={false}>
+			{{
+				trigger: (onToggleOpen) => <Button onClick={onToggleOpen}>Own Width</Button>,
+				content: (onClose) => (
+					<div
+						onpointerleave={onClose}
+						styles={{ background: 'orange', width: '350px', fontSize: '32px' }}
+					>
+						My Width is 150px!
```
Comment (aciccarello) [inline src/examples/src/widgets/popup/SetWidth.tsx]: Should this match the div styles?
```suggestion
						My Width is 350px!
```

File: src/popup/index.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+			width: matchWidth ? `${triggerSize.width}px` : 'auto',
+			left: `${triggerPosition.left}px`,
+			opacity: '1'
+		};
+
+		if (position === 'below') {
+			if (willFit.below) {
+				wrapperStyles.top = `${triggerBottom}px`;
+			} else {
+				wrapperStyles.top = `${triggerTop - wrapperDimensions.size.height}px`;
```
Comment (aciccarello) [inline src/popup/index.tsx]: Do we need to handle when it won't fit above or below? I'm thinking this could happen with a mobile phone on landscape.

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-8kzm1fp03.now.sh
* Tests Deployment: https://dojowidgets-kwyrutkqe.now.sh

File: src/examples/src/widgets/popup/SetWidth.tsx (hunk 1 above)
Comment (tomdye) [reply src/examples/src/widgets/popup/SetWidth.tsx]: yup, oops

File: src/popup/index.tsx (hunk 2 above)
Comment (tomdye) [reply src/popup/index.tsx]: I plan for us to tackle left / right positions as a follow up PR / issue

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-gwvymm8nr.now.sh
* Tests Deployment: https://dojowidgets-kwyrutkqe.now.sh

File: src/popup/index.tsx (hunk 2 above)
Comment (vansimke) [reply src/popup/index.tsx]: Should we get an issue posted just so we don't lose track of this?

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-knnvn42lv.now.sh
* Tests Deployment: https://dojowidgets-47qewqqkp.now.sh

File: src/popup/index.tsx (hunk 2 above)
Comment (tomdye) [reply src/popup/index.tsx]: I added it to the existing issue already @vansimke

**Pull request merged**